easybuild/framework/easyconfig/types.py

```diff
@@ -495,10 +500,17 @@ def ensure_iterable_license_specs(specs):
     'opt_keys': [],
     'req_keys': ['files', 'dirs'],
 }))
-CHECKSUMS = (list, as_hashable({'elem_types': [STRING_OR_TUPLE_LIST]}))
-
-CHECKABLE_TYPES = [CHECKSUMS, DEPENDENCIES, DEPENDENCY_DICT, TOOLCHAIN_DICT, SANITY_CHECK_PATHS_DICT,
-                   STRING_OR_TUPLE_LIST, TUPLE_OF_STRINGS]
+CHECKSUM = (list, as_hashable({'elem_types': [str, TUPLE_OF_STRINGS]}))
```

Comment:
When can a single checksum be a list of strings? It's either a string or a tuple of strings, not a list?

Reply:
Not sure what to change this to...?

Reply:
So, an element in `checksums` can be either:

(1) a single string
(2) a single 2-tuple with (<type>, <checksum>)
(3) once #2958 is merged, a single n-tuple with alternative checksums
(4) a list of checksums that should *all* pass (where each checksum can be in formats 1-3)
(5) a dict with filename-to-checksum mappings (this PR)

So, with that in mind, I'm not sure if the original `CHECKSUMS` here actually was correct, since that only (partially) covers (4)?

I need to check this again when I'm more awake. :)